src/Kernel/BlockClosure.class.st

```diff
@@ -341,10 +341,27 @@ BlockClosure >> ifError: errorHandlerBlock [
 				ifFalse: [self error: err]]
 "
 
+	self deprecated: 'Please use ifErrorDo: instead'.
+
 	^ self on: Error do: [:ex |
 		errorHandlerBlock cull: ex description cull: ex receiver]
 ]
 
+{ #category : #evaluating }
+BlockClosure >> ifErrorDo: errorHandlerBlock [
+	"Evaluate the block represented by the receiver, and normally return it's value.  If an error occurs, the errorHandlerBlock is evaluated, and it's value is instead returned.  The errorHandlerBlock must accept zero or one parameter (the error object)."
+	"Examples:
+		[1 whatsUpDoc] ifErrorDo: [:err | 'huh?'].
+		[1 / 0] ifErrorDo: [:err |
+			ZeroDivide = err class
+				ifTrue: [Float infinity]
+				ifFalse: [self error: err]]
```

Comment:
Eventually, there could be executable examples created from this comment by using #>>> method.